src/core/reference/include/ngraph/runtime/reference/topk.hpp

```diff
@@ -100,9 +99,9 @@ void topk(const T* arg,
             break;
```

Comment:
```
case op::v1::TopK::SortType::SORT_INDICES:
            std::sort(workspace.begin(), workspace.begin() + k, sort_indices_ascending<T, U>);
```
According to revisions in latest spec [TopK_1](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/sort/TopK_1.md), [TopK_3](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/sort/TopK_3.md) and [TopK_11](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/sort/TopK_11.md),

- **_mode=max, sort=index - descending by index_**

the implement here is not always right. If the above spec revision is what we want, then we should align the implements. Please also refer to https://github.com/openvinotoolkit/openvino/pull/15583#discussion_r1133032295.

cc @tomdol

Reply:
I changed it accordingly and now CPU and Template Plugin Func tests fail.

Reply:
Discussed with Tomasz at https://github.com/openvinotoolkit/openvino/pull/15583#discussion_r1133727732, the revision in spec will be reverted. So the original implement here, always applying ascending sorting for SORT_INDICES is right. No revision is needed. Thanks!

cc @tomdol